.github/workflows/updater-check.yaml

```diff
@@ -0,0 +1,61 @@
+name: Start Updater
+on:
+  push:
+    branches:
+    - 'yli3/**'
+  schedule:
+  - cron: '0 */3 * * *'
+
+jobs:
+  read-versions:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.output-matrix.outputs.matrix }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Read versions
+      id: versions
+      run: |
+        VERSIONS=$(cat scanner/updater/version/RELEASE_VERSION)
+        echo "$VERSIONS" | grep -Eo '([0-9]+\.[0-9]+\.[0-9]+|[0-9]+\.[0-9]+\.x-[0-9]+-g[a-f0-9]+)' > versions.txt
+
+    - name: Generate matrix JSON
+      id: output-matrix
+      run: |
+        EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+        MATRIX=$(jq -Rs '{ versions: split("\n")[:-1] }' versions.txt)
+        echo "matrix<<EOF"$'\n'"$MATRIX"$'\n'EOF >> $GITHUB_OUTPUT
+        echo "$MATRIX"
+
+  output-versions:
+    needs: read-versions
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: ${{fromJson(needs.read-versions.outputs.matrix).versions}}
+    env:
+      ROX_PRODUCT_VERSION: ${{ matrix.version }}
+    steps:
+    - name: Authenticate with Google Cloud
+      uses: google-github-actions/auth@v1
+      with:
+        credentials_json: ${{ secrets.GOOGLE_SA_CIRCLECI_SCANNER }}
+
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@v1
+
+    - name: Download and unzip file
+      run: |
+        DOWNLOAD_URL="https://github.com/stackrox/stackrox/archive/refs/tags/${{ env.ROX_PRODUCT_VERSION }}.zip"
```

Comment:
Instead of downloading & unzipping the repository, consider checking it out. Two reasons:
- Unauthenticated downloads are rate limited, checkouts with the `actions/checkout@v3` are authenticated, so should be higher rate limit
- the `cd` doesn't always work super nice in GHA, so could be avoided once

Reply:
I tried `git clone --depth 1 --branch ${{ env.ROX_PRODUCT_VERSION }} https://github.com/stackrox/stackrox.git` but it didn't work for my personal branch (version 4.0.x-295-gad49b77433). I am guessing the clone command should be working for released version in the future but I still want to keep it for now.

Reply:
Where does the `4.0.x-295-gad49b77433` tag come from? It does not exist in [this list](https://github.com/stackrox/stackrox/tags)  or `git ls-remote --tags`. Is that maybe a manual tag that was deleted later? Github seems to keep the zip archive around at the endpoint you download it from. I'd rather say that this is a bug and we should not expect deleted tags to be still available there.

Here is the branch I played around with: https://github.com/stackrox/stackrox/compare/yli3/scannerUpdateSupportedVersion...tm/test-scanner-updater-checkout?expand=1. 
Feel free to cherry-pick from there. 

I noticed that nightly tags are excluded in your regex, is that intentional?

Reply:
do you mean this regex?
`echo "$VERSIONS" | grep -Eo '([0-9]+\.[0-9]+\.[0-9]+|[0-9]+\.[0-9]+\.x-[0-9]+-g[a-f0-9]+)' > versions.txt`
I will include that in the fix

Reply:
yes

Reply:
Actually I am not entirely sure if we care about nightly tags, I will leave it to @RTann and @jvdm